adapter/authzOpa/authzOpa_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authzOpa
+
+// NOTE: This test will eventually be auto-generated so that it automatically supports all CHECK and QUOTA
+//       templates known to Mixer. For now, it's manually curated.
+
+import (
+	"context"
+	"testing"
+
+	rpc "github.com/googleapis/googleapis/google/rpc"
+
+	"istio.io/mixer/adapter/authzOpa/config"
+	"istio.io/mixer/pkg/adapter/test"
+	"istio.io/mixer/template/authz"
+)
+
+var (
+	policy = `package mixerauthz
+	    policy = [
+	      {
+	        "rule": {
+	          "verbs": [
+	            "storage.buckets.get"
+	          ],
+	          "users": [
+	            "bucket-admins"
+	          ]
+	        }
+	      }
+	    ]
+	
+	    default allow = false
+	
+	    allow = true {
+	      rule = policy[_].rule
+	      input.user = rule.users[_]
+	      input.verb = rule.verbs[_]
+	    }`
+	checkMethod = "data.mixerauthz.allow"
+	cfg         = &config.Params{
+		Policy:      policy,
+		CheckMethod: checkMethod,
+	}
+)
+
+func TestAccepted(t *testing.T) {
+	info := GetInfo()
+
+	if !contains(info.SupportedTemplates, authz.TemplateName) {
+		t.Error("Didn't find all expected supported templates")
+	}
+
+	b := info.NewBuilder().(*builder)
+	b.SetAdapterConfig(cfg)
+	if err := b.Validate(); err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	handler, err := b.Build(context.Background(), test.NewEnv(t))
+	if err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	authzHandler := handler.(authz.Handler)
+	instance := authz.Instance{
+		Subject: make(map[string]interface {
+		}),
+		Resource: make(map[string]interface {
+		}),
+		Verb: make(map[string]interface {
+		}),
+	}
+	instance.Subject["user"] = "bucket-admins"
+	instance.Verb["verb"] = "storage.buckets.get"
+
+	result, err := authzHandler.HandleAuthz(context.Background(), &instance)
+	if err != nil {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+
+	if result.Status.Code != int32(rpc.OK) {
+		t.Errorf("Got error %v, expecting success", err)
+	}
+}
+
+func TestRejectedRequest(t *testing.T) {
+	info := GetInfo()
+
+	if !contains(info.SupportedTemplates, authz.TemplateName) {
+		t.Error("Didn't find all expected supported templates")
+	}
+
+	b := info.NewBuilder().(*builder)
+	b.SetAdapterConfig(cfg)
+	if err := b.Validate(); err != nil {
+		t.Errorf("Got error %v, expecting success", err)
```

Comment:
Here and in the error check below you should probably use `t.Fatal` since we can't proceed with the test given the broken state.

Reply:
Applied